### ISSUE TITLE
UPGD metrics[0] (mean_loss) is multiplied by the slow-simplex gradient budget and reads 0.0 under step2_strict_digit_readout_default, hiding task-switch error

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#374]
## Summary

`UPGDUpdateResult.metrics` is documented as `[mean_loss, mean_utility, min_utility, max_perturbation_magnitude]` and is the only per-step loss `run_upgd_arrays` / `run_upgd_loop` emit. In `UPGDLearner.update` (`alberta_framework/core/upgd.py`) the two-timescale simplex readout multiplies the returned `loss_value` by `slow_simplex_gradient_scale` — both in `loss_and_aux_fn` (`loss = slow_simplex_gradient_scale * mse_loss`) and in the direct-MSE branch (`loss_value = slow_simplex_gradient_scale * 0.5 * jnp.sum(sq_masked) / denom`). That scale is `readout_slow_simplex_gradient_multiplier`, documented as a *gradient budget* knob ("allocates the bounded gradient budget to the fast simplex branch"). The shipped factory `UPGDLearner.step2_strict_digit_readout_default` sets it to `0.0`, so once the adaptive simplex gate opens on persistent simplex targets, `metrics[0]` reads exactly `0.0` regardless of how wrong the predictions are. Internal gates use `step_mse` and are unaffected, so nothing else surfaces the error.

## Falsifier (deterministic, pristine `main` @ `45279f3`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.upgd import UPGDLearner
learner = UPGDLearner.step2_strict_digit_readout_default(n_heads=3, hidden_sizes=(8,))
state = learner.init(4, jr.key(0)); obs = jnp.ones(4)
A, B = jnp.array([1., 0., 0.]), jnp.array([0., 0., 1.])
for t in range(260):
    tgt = A if t < 200 else B                       # one task switch at t=200
    r = learner.update(state, obs, tgt); state = r.state
    if t in (199, 200, 201, 205, 220):
        print(t, "metrics[0] =", float(r.metrics[0]), " true 0.5*SE =", round(float(0.5*jnp.sum((r.predictions-tgt)**2)), 6))
```

Output on `main`:

```
199 metrics[0] = 0.0  true 0.5*SE = 4.1e-05
200 metrics[0] = 0.0  true 0.5*SE = 0.989012
201 metrics[0] = 0.0  true 0.5*SE = 0.976516
205 metrics[0] = 0.0  true 0.5*SE = 0.580911
220 metrics[0] = 0.0  true 0.5*SE = 0.006874
```

The reported loss is a flat zero across the task switch — a learning curve, plasticity, or forgetting number read from `metrics[:, 0]` for this factory erases exactly the spike a continual-learning lane exists to measure and looks like a perfect learner.

Expected: the gradient-budget scale shapes only the differentiated objective; `metrics[0]` reports the unscaled prediction loss (with the configured target weights and normalization).

## Dedup

`gh issue/pr list --search "mean_loss" / "slow_simplex" / "gradient multiplier" / "step2_strict_digit"`: nothing on this surface. Adjacent, closed: #33 (spread estimators), #46 (sigma0 reduction pin). No open PR touches `core/upgd.py`.

Fix PR follows (failing-test-first).
